File: Jasmine.FastenerDepartment.Wpf/Jasmine.FastenerDepartment.Application/Models/Synchronization/SynchronizationHistoryEntryItem.cs
using Jasmine.FastenerDepartment.Domain.HistoryEntries.Models;

namespace Jasmine.FastenerDepartment.Application.Models.Synchronization;

/// <summary>
/// Synchronization history entry item.
/// </summary>
public class SynchronizationHistoryEntryItem
{
    /// <summary>
    /// Product identifier.
    /// </summary>
    public Guid ProductId { get; set; }

    /// <summary>
    /// List of product history entries.
    /// </summary>
    public ICollection<ProductHistoryEntry> ProductHistoryEntries { get; set; }
}

File: JasmineFastenerDepartment.Kotlin/app/src/main/java/com/szaytsev/jasminefastenerdepartment/network/models/SynchronizationHistoryEntryItem.kt
package com.szaytsev.jasminefastenerdepartment.network.models

import com.szaytsev.jasminefastenerdepartment.data.models.ProductHistoryEntry
import kotlinx.serialization.Serializable

@Serializable
data class SynchronizationHistoryEntryItem(
    val productId: String,
    val productHistoryEntries: List<ProductHistoryEntry>
)